app/models/legal_aid_application_transaction_type.rb

```diff
@@ -4,4 +4,15 @@ class LegalAidApplicationTransactionType < ApplicationRecord
 
   scope :credits, -> { includes(:transaction_type).where(transaction_types: { operation: :credit }) }
   scope :debits, -> { includes(:transaction_type).where(transaction_types: { operation: :debit }) }
+
+  after_commit :cascade_delete_cash_transactions, on: :destroy
```

Comment:
I'd like to pair to better understand the relationships between legal aid applications, transactions, and transaction types.

It feels strange to define a custom `after_commit`, rather than relying on `dependent: :destroy` relationship options and `on_delete: :cascade` database constraints.

I wonder if there's some refactoring we could do to simplify the relationships, but it might not be straightforward.

Reply:
there is no join table via which we can cascade the delete using a `dependant: :destroy` that I can see. Happy to discuss as this was not something I was happy with either.